server/src/main/java/io/crate/metadata/view/ViewInfoFactory.java

```diff
@@ -79,13 +78,10 @@ public ViewInfo create(RelationName ident, ClusterState state) {
             }
             columns = collectedColumns;
         } catch (Exception e) {
-            if (e instanceof ResourceUnknownException) {
-                // Return ViewInfo with no columns in case the statement could not be analyzed,
-                // because the underlying table of the view could not be found.
-                columns = Collections.emptyList();
-            } else {
-                throw e;
-            }
+            // Statement could not be analyzed, because the referenced table either not found
+            // or has been updated and view definition became incompatible with the new schema (https://github.com/crate/crate/issues/14377).
+            columns = Collections.emptyList();
+
```

Comment:
Note, that information schema would be working and "corrupted" view won't be displayed. 

Drawback of such solution (including ignoring `RelationUnknown` exception on master) is having unusable views in the cluster state.  `select * from outdated_view` would raise the same casting error. 

Another ways to fix it:

1. Keep views in sync with tables. I'm not sure it's possible and practical in all scenarios. In the reported issue user wanted to subtract an interval from a timestamp and expected to get a timestamp. We can force view to return array with a single value having the "expected" result but such view can, in turn, break other user queries, dependant on that view.

2.  Ignore all exceptions but also drop the corrupted view as it's anyway unusable. It can also break user's dependant queries (also changing cluster state when accessing information schema is not really safe).

3. Ignore exceptions but maybe log a warning on information schemas queries (view X is outdated and cannot be accessed) and show a better error message on `select * from outdated_view`?

Reply:
> Drawback of such solution (including ignoring RelationUnknown exception on master) is having unusable views in the cluster state.

If it can be fixed e.g. by `CREATE OR REPLACE VIEW ....` I think that is fine.

ad 1) I think that is also bad in table replacement scenarios that we rely on, as reindexing is sometimes needed
ad 2) see 1

ad 3) Instead of a log message (which is limited to host admins) I would rather have an error message in `information_schema.views`

Reply:
> have an error message in information_schema.views

had a concern that PG doesn't have anything like that but we already have our custom `owner` column and in general custom column should be fine in terms of compatibility. Sounds fine to  me, @matriv, what do you think?

Reply:
I think that this PR's solution is the way to go. The view is erroneous when you try to query it and then you can still see what is the query stored from the view through `information_schema.views` and `pg_views` and fix it.

- I wouldn't add extra cols to `information_schema.views` which is SQL standard.
- Adding a log message when querying those catalog info tables, I think it's not helpful for the user.

I have another suggestion, how about adding a comment like:
``-- Corrupeted - needs fix --`` on top of the query of the view, returned in those tables?

Reply:
> I have another suggestion, how about adding a comment like:
-- Corrupted - needs fix -- on top of the query of the view, returned in those tables?

Will do, thanks! Should we mention a better error message on broken queries in the entry?

Reply:
Thanks, added a hint and updated a change entry.